content/docs/guides/verify-jwt.md

```diff
@@ -0,0 +1,124 @@
+
+The `issuer` and `audience` parameters should both be set to the domain of the **upstream application** without the prefixed protocol (for example, `httpbin.corp.example.com`).
+
+<b>Note:</b> We strongly recommend that you explicitly define the expected issuer and audience claims. Relying on a TOFU policy is dangerous in ephemeral serverless environments (such as AWS Lamda or Cloud Run), where applications are typically short-lived.
+
+  </div>
+</details>
+
+### PomeriumVerifier reference
+
+The `PomeriumVerifier` class is the easiest way to verify JWTs. See the reference below for more information:
+
+#### Parameters
+
+| Parameters | Description | Value |
+| :-- | :-- | --- |
+| issuer | The domain of the upstream application (for example, `httpbin.corp.example.com`). | String |
+| audience | The same value as `issuer`. | String |
+| expirationBuffer | Adds padding in seconds to prevent throwing errors for expired JWTs that may have differing server times. Defaults to `0` | Integer |
+| firstUse | Decides whether or not to trust the first JWT. | Boolean |
+| jwtData | The JSON payload containing JWT claims. | Object |
+| verifiedJwtData | The verified JSON payload containing JWT claims. | Object |
+
+#### Methods
+
+| Method | Description |
+| :-- | :-- |
+| getClientJwt | Fetches client JWT from the `/.pomerium/jwt` endpoint. |
+| parseJWT | Decodes JWT token. |
+| getJWKsData | Fetches JWKs data from the `/.well-known/pomerium/jwks.json` endpoint. |
+| verifyPomeriumJWT | Verifies JWT using the `jwt`, `authenticateBaseUrl`, `issuer`, and `audience` parameters. |
+| withHttps | Prepends the URL with the `https://` protocol. |
+| signOut | Signs user out and redirects them with the `/.pomerium/sign_out` endpoint. |
+
+## Manual verification
+
+Though you will likely verify signed headers programmatically in your application's middleware with a third-party JWT library, if you are new to JWT it may be helpful to show what manual verification looks like.
+
+1. Provide Pomerium with a base64-encoded Elliptic Curve ([NIST P-256](https://csrc.nist.gov/csrc/media/events/)) Private Key. In production, you'd likely want to get these from your key management service (KMS).
+
+   ```bash
+   openssl ecparam -genkey -name prime256v1 -noout -out ec_private.pem
+   openssl ec -in ec_private.pem -pubout -out ec_public.pem
+   # careful! this will output your private key in terminal
+   cat ec_private.pem | base64
+   ```
+
+   Copy the base64 encoded value of your private key to Pomerium's environmental configuration variable `SIGNING_KEY`.
+
+   ```bash
+   SIGNING_KEY=LS0tLS1CRUdJTiBQUklWQVRFIEtFWS0tLS0tCk1JR0hBZ0VBTUJNR0J5cUdTTTQ5QWdFR0NDcUdTTTQ5QXdFSEJHMHdhd0lCQVFRZzBQdk1VeVZGeWxTbGZ3eDgKSDBxMUVyOHZlOXBnY3ZzNkV6ZnR5OHlxNnFLaFJBTkNBQVM5ZC96TC9aSXd5ZGQ1RXZMb0xGMytHblVIUS9wdQpQaU45NDV1Y1RpTFRqMDhZalo3U0NJV2JHc2tiK0RIMzJ2aUc2KzRnb0FvWlFUM1R6b2kzRVl6OAotLS0tLUVORCBQUklWQVRFIEtFWS0tLS0tCg==
```

Comment:
```suggestion
   SIGNING_KEY=...
```

people may copy-paste a canned shared key from docs and forget about it.

Reply:
Updated.